testsuite/tests/letrec-disallowed/float_block_disallowed.ml

```diff
@@ -1,18 +1,35 @@
 (* TEST
    * flat-float-array
-   ** toplevel
+   ** expect
 *)
 
 (* when the -flat-float-array optimization is active (standard in
    OCaml versions up to at least 4.07), creating an array inspects its
    first element to decide whether it is a float or not; it would thus
-   be unsound to allow to recursively define a float value and an
+   be unsound to allow to recursively define avalue and an
    array starting with that element (in general we disallow using a
    recursively-defined value in an array literal).
```

Comment:
> `(in general we disallow using a recursively-defined value in an array literal).`

This sentence (from a previous PR) is not correct: see above.  It could be fixed by adding something along the lines of "unless the value is known to have a type not compatible with float".